### PR TITLE
refactor: unify error metrics

### PR DESCRIPTION
**Pull request by StarpTech**
## Motivation and Context

This PR refactors error metrics. We ensure that `wg.reqeust.error` is added to all metrics except request, response in case of an error. For this, we had to move metric collection from the transport to the engine loader hooks where we know it is a GraphQL error.

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .run/router dev.run.xml [hunk 1]
```diff
@@ -2,9 +2,9 @@
   <configuration default="false" name="router dev" type="GoApplicationRunConfiguration" factoryName="Go Application">
     <module name="cosmo" />
     <working_directory value="$PROJECT_DIR$/router" />
-    <go_parameters value="-race" />
```
Comment (Aenimus) [inline .run/router dev.run.xml:5]: Are both this subtraction and addition intended?

File: router/config.yaml [hunk 2]
```diff
@@ -3,4 +3,14 @@
 # See pkg/config/config.go for the full list of configuration options.
 # This file is used to debugging purposes only.
 
-version: "1"
\ No newline at end of file
+version: "1"
```
Comment (Aenimus) [inline router/config.yaml:6]: Did you intend to commit this?

**Review by Aenimus — APPROVED** [review]
LGTM but see comments

File: .run/router dev.run.xml (hunk 1 above)
Comment (StarpTech) [reply .run/router dev.run.xml:5]: No, thanks.

**Pull request merged**